src/NUnitFramework/tests/Api/FrameworkControllerTests.cs

```diff
@@ -246,38 +262,55 @@ public void ConstructController()
         [Test]
         public void LoadTestsAction_GoodFile_ReturnsRunnableSuite()
         {
-            new FrameworkController.LoadTestsAction(_controller, _handler);
-            var result = TNode.FromXml(_handler.GetCallbackResult());
-
-            Assert.That(result.Name.ToString(), Is.EqualTo("test-suite"));
-            Assert.That(result.Attributes["type"], Is.EqualTo("Assembly"));
-            Assert.That(result.Attributes["id"], Is.Not.Null.And.StartWith("ID"));
-            Assert.That(result.Attributes["name"], Is.EqualTo(EXPECTED_NAME));
-            Assert.That(result.Attributes["runstate"], Is.EqualTo("Runnable"));
-            Assert.That(result.Attributes["testcasecount"], Is.EqualTo(MockAssembly.Tests.ToString()));
-            Assert.That(result.SelectNodes("test-suite").Count, Is.EqualTo(0), "Load result should not have child tests");
+            var originalPrefix = Test.IdPrefix;
+            var controller = new FrameworkController(MOCK_ASSEMBLY_PATH, TEST_PREFIX, _settings);
+            try
+            {
+                new FrameworkController.LoadTestsAction(controller, _handler);
+                var result = TNode.FromXml(_handler.GetCallbackResult());
```

Comment:
As little as possible should be in the `try` block. It matches the exact relevance of the `finally` block to the contents of the `try` block, which is good for clarity. Could you end the `try` right after `_handler.GetCallbackResult()`?

Reply:
I wasn't sure where the cut off was, so I put it across the entire method, as it's more clear to someone looking at the method what it was doing (at least, in my opinion).

Reply:
All those Assert statements are running with the wrong `Test.IdPrefix`. Could you move at least those out of the `try`?